core/include/scipp/core/except.h

```diff
@@ -139,6 +149,11 @@ template <class T> void unit(const T &object, const units::Unit &unit) {
   expect::equals(object.unit(), unit);
 }
 
+template <class T>
+void unit_one_of(const T &object, std::initializer_list<units::Unit> units) {
```

Comment:
Name these `.._any_of`, for consistency with `std::any_of`, which does a similar thing??